_posts/2022-06-26-contextual-slu.md

```diff
@@ -0,0 +1,68 @@
+semantic frames. The problem of contextual SLU majorly focuses on effectively incorporating dialogue information.
+
+
+## Why context?
+There are many in-scope intents that contain a lot of contextual information within the bot prompt and/or by using State-level information. Examples from one of our clients are below.
+|                                                                                 **Bot Prompt**                                                                                | **User Response** |        **Intent**        |
+|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|:-----------------:|:------------------------:|
+| Hi! I am Divya, your Hathway virtual assistant. Are you looking for a new Hathway Broadband connection?                                                                       | yes** **          | confirm_new_connection   |
+| Alright! To check your internet connectivity status, the device should be connected to the power supply and switched on. Will it be possible for you to switch on the device? | yes               | confirm                  |
+| Okay!. To start with, please tell me if you are next to your Hathway device?                                                                                                  | I am              | confirm_near_device** ** |
+
+
+Having that extra context from the bot can definitely help us to make a better contextual model.
+
+
+## Using Bot prompts as context
+
+We curate a private data of our clients wherein we collect user utterances along with all the bot prompts. Earlier, our systems
+were dependent on solely using user utterances to build our Intent classifier but after conctenating bot prompts to our utterances, we
+observed around 30% jump in our intent-F1 scores for some of our clients.
+
+## Some probable approaches from literature
+
+After observing an improvement in our models by just using a single bot prompt, we decided to a delve a bit further and found out many 
+approaches that can be utilized for our use-case. We discuss a few of them below:
+
+### Encoding dialogue History [1, 2]
+1. We can encode the complete dialogue history as shown below. Let us assume that the dialogue is
+a sequence of $D_{t} = {u_{1}, u_{2}.. u_{t}}$ bot and user utterance and at every time 
+step $t$ we are trying to output the classification for the user utterance $u_{t}$, given $D_{t}$.
+We then divide the model into 2 components, the context encoder that acts on $D_{t}$ to produce
+a vector representation of the dialogue context denoted by $h_{t} = H(D_{t})$ and the tagger, which takes
+this context encoding $h_{t}$, and the current utterance $u_{t}$ as input and produces the intent output.
+
+    <img src= '../assets/images/contextual/encoder_context.png' alt='drawing' width='500'>
```

Comment:
The architecture image isn't clear. Can you annotate this better ?

Reply:
Will increase the image size and annotate it better